src/client/resources/web/js/chat.mjs

```diff
@@ -56,29 +56,48 @@ function storeMessage(json) {
 
         localStorage.setItem('chatMessagesJSON', JSON.stringify(messages));
     } catch (e) {
-        console.warn('Failed to store message:', e);
+        console.error('Failed to store message:', e);
     }
 }
 
 /**
  * Add a message to the chat.
- * @param {Component} json
- * @param {boolean} store
+ * @param {string} rawJson
  */
-function addMessage(json, store = true) {
-    console.log(json);
+function displayMessage(rawJson) {
+    console.log(rawJson);
     const div = document.createElement('div');
     div.className = 'message';
-    div.innerHTML = parseMinecraftText(json);
+
+    /** @type {unknown} */
+    const json = JSON.parse(rawJson);
```

Comment:
I feel like this would fail. While the variable is called rawJSON it is already parsed and an object when received through websockets.

Trying to parse a js object as JSON again will throw a syntax error.